engine/src/in_joystick.cpp

```diff
@@ -113,15 +113,6 @@ void myGlutJoystickCallback( unsigned int buttonmask, int x, int y, int z )
     }
 }
 
-JoyStick::JoyStick()
-{
-    for (int j = 0; j < MAX_AXES; ++j) {
-        axis_axis[j]    = -1;
-        axis_inverse[j] = false;
-        joy_axis[j]     = axis_axis[j] = 0;
-    }
-    joy_buttons = 0;
-}
```

Comment:
Hmm. I don't think getting rid of the JoyStick constructor is the right approach. Let's try something else.

I take it that you are trying to clean up the output from the program when no joystick is present?

Reply:
This PR does a few independent things / cleanups:
- getting rid of an apparently unused constructor
- getting rid of a useless console message
- getting rid of a duplicated extern definition

I separated each in its own commit

This constructor looks unused, the only one used is the one taking a `JoyStick::JoyStick(int)` parameter.

I may be missing something, though...

Reply:
I would say that if a default constructor has already been implemented, it's generally a good idea to leave it, even if it does not appear to be used anywhere at the moment. Future-proofing, you might call it.